folly/m4/fb_filter_pkg_libs.m4

```diff
@@ -2,7 +2,7 @@ AC_DEFUN([FB_FILTER_PKG_LIBS],
   [AC_REQUIRE([AC_PROG_SED])
    deps_=`for p in $PKG_DEPS; do pkg-config --libs $p; done`
    filter_=`for l in $deps_;dnl
-     do echo $l | $SED -e 's%\(-l.*\)%-e s/\1//%' -e '/^-L/d';dnl
+     do echo $l | $SED -e 's%\(-l.*\)%-e s/\1//%' -e '/^-L/d' -e '/^-W/d';dnl
```

Comment:
It would be more general to filter all options except the -l we want to process. I.E.:

    do echo $l | $SED -n '-e 's%\(-l.*\)%-e s/\1//p%'